library/index.c

```diff
@@ -530,10 +530,14 @@ int eblob_generate_sorted_index(struct eblob_backend *b, struct eblob_base_ctl *
 	struct eblob_map_fd src, dst;
 	int fd, err, len;
 	char *file, *dst_file;
+	int cache_is_empty = 0;
 
 	if (b == NULL || bctl == NULL)
 		return -EINVAL;
 
+	/* check that eblob cache is empty. If cache is empty we can skip iterating and flushing it */
+	cache_is_empty = eblob_cache_empty(b);
+
```

Comment:
> Not at all. Empty cache is valid at any moment - we just have sorted a blob and removed all data from cache because of that.

Ok, while all our blobs have sorted indexes we have empty cache. But while all our blobs have sorted indexes there is no blob which need to be sorted.

>Again, this is not the price for all these questions and a risk of data corruption.

I do not see any risk of data corruption from these changes.